src/global_logic.js

```diff
@@ -598,14 +598,24 @@ module.exports.calcBasedOneSummon = function (summonind, prof, buff, totals) {
             var LBCriticalArray = getLBCriticalArray(totals[key]["LB"]);
             var EXLBCriticalArray = getEXLBCriticalArray(totals[key]["EXLB"]["Critical"]);
             var normalOtherCriticalBuffArray = totals[key]["normalOtherCriticalBuff"];
-            var normalOtherCriticalArray = totals[key]["normalOtherCritical"].concat(LBCriticalArray, EXLBCriticalArray, normalOtherCriticalBuffArray);
+            var normalOtherCriticalArray = totals[key]["normalOtherCritical"].concat(LBCriticalArray, EXLBCriticalArray, normalOtherCriticalBuffArray, totals[key]["criticalBuff"], buff["criticalBuff"]);
 
             var criticalArray = module.exports.calcCriticalArray(totals[key]["normalCritical"], totals[key]["magnaCritical"], normalOtherCriticalArray, totalSummon);
-            var criticalRatio = module.exports.calcCriticalRatio(criticalArray)
+            var criticalRatio = module.exports.calcCriticalRatio(criticalArray);
         } else {
             var damageUP = 0.0;
-            var criticalArray = {};
-            var criticalRatio = 1.0
+            if (prof.retsujitsuNoRakuen) {
```

Comment:
Can you consider ` else if (prof.enemyElement == "non-but-critical" || prof.retsujitsuNoRakuen) {`
to reduce the clone code.